elastic/kibana
# [ResponseOps][mget] Poll for tasks less frequently when the task load doesn't need it

Resolves https://github.com/elastic/kibana/issues/196584

## Summary

This PR updates the task poll interval logic for projects using the mget strategy to optimize request loads to Elasticsearch, particularly for smaller projects with low utilization. When task manager (TM) utilization is below 25%, the poll interval will be set to 3 seconds instead of the current 500 milliseconds. This change does not affect projects utilizing `update_by_query`.

The existing backpressure logic remains unchanged for handling errors. The only adjustment occurs in scenarios where there are no errors, the TM utilization is below 25%, and the poll interval is less than 3 seconds. In such cases, the poll interval will increase to 3 seconds, even if the backpressure logic has not fully reset the interval to its original value.

I just chose 25%, but I am definitely open to other ideas.


### Checklist



### To verify

- Start Kibana and go to `http://localhost:5601/api/task_manager/_health` and verify the poll interval is 3s
- Create some alerting rules scheduled to run every second, and let them run. I created 4 rules.
- Check `http://localhost:5601/api/task_manager/_health` again to verify that with rules running the poll interval is back to 500ms. (It may take a couple refreshes for the health api to reflect the changes)
